lighthouse-cli/test/smokehouse/test-definitions/screenshot/expectations.js

```diff
@@ -12,9 +12,39 @@ const expectations = [
   {
     artifacts: {
       FullPageScreenshot: {
-        width: '>1000',
-        height: '>1000',
-        data: /data:image\/jpeg;base64,.{10000,}$/,
+        screenshot: {
+          width: '>1000',
+          height: '>1000',
+          data: /data:image\/jpeg;base64,.{10000,}$/,
+        },
+        nodes: {
+          'page-0-BODY': {
+            top: 8,
+            bottom: 1008,
+            left: 8,
+            right: 1008,
+            width: 1000,
+            height: 1000,
+          },
+          'page-1-P': {
+            top: 8,
+            bottom: 26,
+            left: 8,
+            right: 1008,
+            width: 1000,
+            height: 18,
+          },
+          '5-2-BODY': {
+            top: 8,
+            bottom: 1008,
+            left: 8,
+            right: 1008,
+            width: 1000,
+            height: 1000,
+          },
+          '5-3-HTML': {},
```

Comment:
the sizes here are env-dependent so I just removed them.

This also illustrates that the same node can get assigned multiple ids. We could do a reverse lookup in getNodeDetailsImpl to dedupe this, but it'd only dedupe within the same pass. Should we?

Reply:
I like the idea of deduping them, it would make the results less confusing. Would it be possible to add a pass number to the lhid?

Reply:
+1 to deduping, I wouldn't love a pass number for FR migration, but this is going to be tricky anyway

Reply:
alternative, if we provide more structure here we could do the de duping when collecting in full page screenshot. i'll experiment.

Reply:
wait no sorry, that'd involve editing the ids in the LHR which we are trying hard not to do.

Reply:
iirc top/left were non constant, which was strange. We aren't losing anything by not having _more_ rect checks here imo.

what I'd really like is a way to mark "no other keys" on an object.